internal/throttler/throttler.go

```diff
@@ -0,0 +1,111 @@
+//go:generate mockgen -source throttler.go -destination ../mocks/mock_throttler.go -package mocks
+
+package throttler
+
+import (
+	"context"
+	"time"
+
+	grpc_ctxtags "github.com/grpc-ecosystem/go-grpc-middleware/tags"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/openfga/openfga/internal/build"
+	"github.com/openfga/openfga/pkg/telemetry"
+)
+
+type Throttler interface {
+	Close()
+	Throttle(context.Context)
+}
+
+type NoopThrottler struct{}
+
+var _ Throttler = (*NoopThrottler)(nil)
+
+func (r *NoopThrottler) Throttle(ctx context.Context) {
+}
+
+func (r *NoopThrottler) Close() {
+}
+
+// DispatchThrottler implements a throttling mechanism that can be used to control the rate of dispatched sub problems in FGA queries.
+// Throttling will start to kick in when the dispatch count exceeds the configured dispatch threshold.
+type DispatchThrottler struct {
```

Comment:
Can you rename this so that we don't have the name "Dispatch" in it?  This is generic enough to be used by other throttling client such as datastore throttling